test/langswitch/LanguageSwitchControllerSpec.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package langswitch
+
+import langswitch.LanguageSwitchController
+import langswitch.Languages.English
+import play.api.http.Status
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{defaultAwaitTimeout, status}
+import support.AppSpec
+
+final class LanguageSwitchControllerSpec extends AppSpec {
+  private val controller = app.injector.instanceOf[LanguageSwitchController]
+
+  "switch to language should return SEE OTHER" in {
+    val fakeRequest = FakeRequest("GET", "/")
+
+    val result = controller.switchToLanguage(English)(fakeRequest)
+
+    status(result) shouldBe Status.SEE_OTHER
+
+  }
+
+  "notFound should lead to not found page" in {
+    val fakeRequest = FakeRequest("GET", "/")
+
+    val result = controller.notFound(fakeRequest)
+
+    status(result) shouldBe Status.OK
```

Comment:
there's an opportunity to test the correct content is shown here - just testing for an OK could let a lot of things happen and still let this test pass.